plugins/deployment_manager/DeploymentManager.ts

```diff
@@ -122,9 +125,19 @@ export class DeploymentManager {
     this.contractsCache = null;
   }
 
+  /* Conditionally executes an action */
+  async idempotent<T>(
```

Comment:
IMO, feels a bit odd to call this fn `idempotent`. I understand it's main use case is to support idempotent deploys, but this is more of a generic helper to conditionally execute some function. Maybe we could name it `conditionalRetry`? 

e.g.
```
await deploymentManager.conditionalRetry(
  async () => !sameAddress($configuratorImpl, configuratorImpl.address),
  async () => {
    debug(`Setting Configurator implementation to ${configuratorImpl.address}`);
    await wait(cometAdmin.connect(admin).upgrade(configurator.address, configuratorImpl.address));
  }
);
```

vs

```
await deploymentManager.idempotent(
  async () => !sameAddress($configuratorImpl, configuratorImpl.address),
  async () => {
    debug(`Setting Configurator implementation to ${configuratorImpl.address}`);
    await wait(cometAdmin.connect(admin).upgrade(configurator.address, configuratorImpl.address));
  }
);
```

Seems somewhat clearer what's happening

Reply:
Yeah it could also be called `maybe`, tbh I didn't think about the naming of this one at all

Reply:
I suggest we do another stylistic search and replace after this if we can agree about some other names for things